src/main/rx/sumd.c

```diff
@@ -53,17 +53,29 @@ bool sumdInit(rxConfig_t *rxConfig, rxRuntimeConfig_t *rxRuntimeConfig, rcReadRa
 {
     UNUSED(rxConfig);
 
+    rxRuntimeConfig->channelCount = SUMD_MAX_CHANNEL;
+
     if (callback)
         *callback = sumdReadRawRC;
 
-    rxRuntimeConfig->channelCount = SUMD_MAX_CHANNEL;
-
     serialPortConfig_t *portConfig = findSerialPortConfig(FUNCTION_RX_SERIAL);
     if (!portConfig) {
         return false;
     }
 
-    serialPort_t *sumdPort = openSerialPort(portConfig->identifier, FUNCTION_RX_SERIAL, sumdDataReceive, SUMD_BAUDRATE, MODE_RX, SERIAL_NOT_INVERTED);
+    portMode_t serialPortMode;
+    portSharing_e serialPortSharing = determinePortSharing(portConfig, FUNCTION_RX_SERIAL);
+    if (serialPortSharing == PORTSHARING_SHARED){
+    	if (portConfig->functionMask & ALL_FUNCTIONS_SHARABLE_WITH_RX_SERIAL)
+            serialPortMode = MODE_RXTX;
+    	else
+    		return false;
+    	}
+    else {
+        serialPortMode = MODE_RX;
+    }
+
+    serialPort_t *sumdPort = openSerialPort(portConfig->identifier, FUNCTION_RX_SERIAL, sumdDataReceive, SUMD_BAUDRATE, serialPortMode, SERIAL_NOT_INVERTED);
```

Comment:
Sorry, commented wrong line, should be at SBUS code. SBUS_PORT_OPTIONS is inverted, TX will be inverted too on 303